### ISSUE TITLE
403 Forbidden when trying to upload PDF as blob to S3 bucket

### ISSUE THREAD

**Issue by laneschmidt** [jbesw/askjames-s3uploader#4]
please look at this Stackoverflow question: https://stackoverflow.com/questions/56709408/403-forbidden-when-trying-to-upload-pdf-as-blob-to-s3-bucket

to be brief, the supplied code results in a `403 Forbidden` error. I'm looking at a possible fix, but am unsure how to get the ContentLength of my file on the frontend to the backend.

**Comment by jbesw** [comment]
Hi, this is indicative of a permissions problem with the IAM policy used by your Lambda function. The function itself must have permission to upload files to the S3 bucket (this is not granted by the basic execution role). If you add S3 write permissions for the target bucket to the IAM role used by the function, this should resolve the problem.

**Comment by jbesw** [comment]
Digging into this, there are two other issues that could be causing your problem:

1. Did you enable CORS on your S3 bucket? In the Permissions --> CORS configuration tab, ensure you have a policy set.

2. Is the bucket set to either:
- Be publicly viewable or
- Is it a bucket that can never be publicly viewable, but you are using the public ACL when uploading the object?
- Do you have an account policy preventing publicly viewable objects but attempting to upload an object with the public ACL?

Basically, you cannot upload objects with a public ACL into a bucket where there is some restriction preventing that - you'll get the 403 error you describe. HTH!

**Comment by jbesw** [comment]
I'm closing this since I believe the issue is resolved but please let me know if that's not the case.
